java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -181,6 +182,22 @@ void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
         programState = programState.stackValue(symbolicValue);
         break;
       }
+      case Opcodes.INVOKESPECIAL:
+      case Opcodes.INVOKESTATIC:
+        org.objectweb.asm.Type methodType = org.objectweb.asm.Type.getMethodType(instruction.fieldOrMethod.desc);
+        boolean isStatic = instruction.opcode == Opcodes.INVOKESTATIC;
+        int arity = isStatic ? methodType.getArgumentTypes().length : methodType.getArgumentTypes().length + 1;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS864)